src/server/handlers/mod.rs

```diff
@@ -53,29 +100,25 @@ impl Handlers {
             )),
         );
 
+        // subnet manager methods
         let pool = Arc::new(SubnetManagerPool::from_reload_config(config));
         handlers.insert(
             String::from(json_rpc_methods::CREATE_SUBNET),
-            HandlerWrapper::CreateSubnet(CreateSubnetHandler::new(pool)),
+            HandlerWrapper::CreateSubnet(CreateSubnetHandler::new(pool.clone())),
+        );
+        handlers.insert(
+            String::from(json_rpc_methods::LEAVE_SUBNET),
+            HandlerWrapper::LeaveSubnet(LeaveSubnetHandler::new(pool.clone())),
+        );
+        handlers.insert(
+            String::from(json_rpc_methods::KILL_SUBNET),
+            HandlerWrapper::KillSubnet(KillSubnetHandler::new(pool.clone())),
+        );
+        handlers.insert(
+            String::from(json_rpc_methods::JOIN_SUBNET),
+            HandlerWrapper::JoinSubnet(JoinSubnetHandler::new(pool)),
         );
 
         Ok(Self { handlers })
     }
-
-    pub async fn handle(&self, method: Method, params: Value) -> Result<Value> {
-        if let Some(wrapper) = self.handlers.get(&method) {
-            match wrapper {
-                HandlerWrapper::CreateSubnet(handler) => {
-                    let r = handler.handle(serde_json::from_value(params)?).await?;
-                    Ok(serde_json::to_value(r)?)
-                }
```

Comment:
I am wondering if the body for each of this branches can be de-duplicated without the need of a macro, it is the same for all of them

Reply:
No need enum and macro now, I found a way to handle it already!